<launch>
  <arg name="launch_prefix" default="" /> <!-- set to value="gdbserver localhost:10000" for remote debugging -->
  <arg name="camera_node_namespace" default="camera" />
  <arg name="camera_name" default="camerav2_1280x720" />
  <arg name="camera_info_dir" default="$(find tag_utils)/camera_info" />
  <arg name="image_topic" default="image_rect" />
  
  <node pkg="tag_utils" type="camera_node" name="$(arg camera_node_namespace)" clear_params="true" output="screen" launch-prefix="$(arg launch_prefix)">
    <!-- Remap topics from those used in code to those on the ROS network -->
    <param name="camera_name" value="$(arg camera_name)" />
    <param name="camera_info_url" value="$(arg camera_info_dir)/$(arg camera_name).yaml" />
  </node>
  
  <arg name="continous_node_namespace" default="apriltag_ros_continuous_node" />
  <arg name="camera_frame" default="camera" />

  <!-- Set parameters -->
  <rosparam command="load" file="$(find apriltag_ros)/config/settings.yaml" ns="$(arg continous_node_namespace)" />
  <rosparam command="load" file="$(find apriltag_ros)/config/tags.yaml" ns="$(arg continous_node_namespace)" />
  
  <node pkg="apriltag_ros" type="apriltag_ros_continuous_node" name="$(arg continous_node_namespace)" clear_params="true" output="screen" launch-prefix="$(arg launch_prefix)" >
    <!-- Remap topics from those used in code to those on the ROS network -->
    <remap from="image_rect" to="$(arg camera_name)/$(arg image_topic)" />
    <remap from="camera_info" to="$(arg camera_name)/camera_info" />
    <param name="remove_duplicates" type="bool" value="true" />
    <param name="camera_frame" type="str" value="$(arg camera_frame)" />
    <param name="publish_tag_detections_image" type="bool" value="true" />      <!-- default: false -->
  </node>

  <arg name="node_namespace" default="tf" />
  <rosparam file="$(find tag_utils)/config/transforms.yaml" />
  <node pkg="tag_utils" type="transform_backend" name="$(arg node_namespace)" clear_params="true" output="screen" launch-prefix="$(arg launch_prefix)">
  </node>

</launch>